### ISSUE TITLE
rmsnorm_pre_all_gather: kernel assumes zero padding but ttnn.ones fills padding with ones

### ISSUE THREAD

**Issue by HalfBloodPrince010** [tenstorrent/tt-metal#43350]
## Description

`ttnn::rms_norm_pre_all_gather` relies on an implicit contract that tile padding is zero-initialized. This assumption is violated when inputs are created with `ttnn.ones` or similar operations that pad value other than zero, resulting in incorrect statistics.

## Background

The kernel (`rmsnorm_pre_allgather.cpp`) computes `E(x²)` over the full padded tile width. It cannot currently distinguish between logical width vs padded width.

For instance, the logical shape `[1, 1, 128, 49]` (width 49 → padded to nearest multiple of TILE_WIDTH = 64 = 2 × 32 tiles)

```
padded_width = ceil(W / TILE_WIDTH) * TILE_WIDTH
```

| Input creation | Column 0 (sum(x²)) | Expected |
|---|---|---|
| `ttnn.from_torch(torch.ones(...))` | **49** | 49 ✓ |
| `ttnn.ones(shape=[1,1,128,49])` | **64** | 49 ✗ |

`ttnn.ones` fills the entire padded tile buffer, i.e14 element 

```
0..49   real data
50..63  padding area
```
with 1s. 

The kernel then includes those padding elements (non-zeroes) in the reduction.

## Problem

The TTNN op `rms_norm_pre_all_gather` expects the input width shape be a multiple of TILE_WIDTH (32)

Suppose input width is:
```
W = 50
```

TT Tile storage
```
ceil(50 / 32) * 32 = 64
```

So the tile has positions:
```
0..49   real data
50..63  padding area
```

From a compiler perspective, a compiler emitting:

```python
ttnn.rms_norm_pre_all_gather(ttnn.ones(shape=[1,1,128, 49], ...))
```
is a perfectly valid op sequence.

But if the padding area has non-zero values (which is the case from ttnn.ones) then the
`sum(input ^ 2)` will produce incorrect stats and doesn't fail.

## Expected Behavior

1. The `rmsnorm_pre_all_gather` kernel should accept the logical width and mask padding in the last tile before reduction. 
- This might not be an issue in `pre_all_gather` ops if the padded values are 0, but for `post_all_gather` which computes mean, should do on the logical width and not the padded width.

or

2. `ttnn.ones` (and similar tensor creation ops) should zero-fill tile padding regions, matching the behavior of `ttnn.from_torch`

or

3. The TTNN op should validate/document the zero-padding precondition and enforce it. 
- Note: This will requires us to maintain information on logical width vs padded width
```
                   x0 ... x49                                      x50         x63
real data          1 ..... 0                           padding     0        ... 0

```
We would need information to distinguish when does the pad value = 0.0 start in the padded width since original
width is also ending with 0.

## Impact

If inputs are created via ops that do not zero-fill tile padding (e.g., `ttnn.ones`, `ttnn.full`) and are passed to rms_norm_pre_all_gather with dimensions not multiples of TILE_WIDTH (32), the kernel will silently compute incorrect sum(x) and sum(x^2) statistics.

**Comment by bbradelTT** [comment]
This should have been fixed for https://github.com/tenstorrent/tt-metal/issues/31983 
There were two PRs merged for that issue over the last two weeks: https://github.com/tenstorrent/tt-metal/pull/42013 and https://github.com/tenstorrent/tt-metal/pull/42549

@HalfBloodPrince010 please try on the latest commit. If this is still not working as expected, please provide a commit hash and a simple test that recreates the issue you are seeing.

**Comment by HalfBloodPrince010** [comment]
Thank you @bbradelTT this should fix the issue as `rmsnorm_pre_all_gather` calls `ttnn::prim::layer_norm` underneath, where the implicit padding happens. Closing the issue.
